2024 and 2015 NSDUH Dataset Analysis <br>
Xavier McFadden | Data Science for Addiction Research Fellowship | Spring 2026 <br>
Exploratory analysis of stimulant and sedative misuse in the past year between ages 16 and 25 using NSDUH datasets (2024 and 2015 respectively) <br>
Due 3/6/26

In [4]:
import pandas as pd
import sys
import os

sys.path.append(os.path.join(os.getcwd(),'..', 'scripts'))
from load_data import load_2024, load_2015, load_2024_full# type: ignore
from analyze import misuse_counts, health_breakdown, distress_correlation

df_filtered2024 = load_2024(os.path.join(os.getcwd(),'..', 'data', 'NSDUH_2024_Tab.txt'))
df_filtered2015 = load_2015(os.path.join(os.getcwd(),'..', 'data', 'NSDUH_2015_Tab.tsv'))
df_full_2024 = load_2024_full(os.path.join(os.getcwd(),'..', 'data', 'NSDUH_2024_Tab.txt'))

print(df_filtered2024.shape)
print(df_filtered2015.shape)


(18011, 9)
(19131, 6)


2024 Dataset (ages 16-25): <br>
Stimulant misuse in the past year: <br>
317 + 91 = 408 <br>
Sedative misuse in the past year: <br>
16 + 29 = 45

2015 Dataset Analysis: <br>
Stimulant misuse in the past year: <br>
346 + 873 = 1219 <br>
Sedative misuse in the past year: <br>
41 + 104 = 145 <br>

2015 -> 2024 <br>
 <br>
Stimulant misuse in the past year: <br>
1219 -> 408 <br>
Sedative misuse in the past year: <br>
145 -> 45 <br>
<br>
Overall, misuse in both categories has decreased by a factor of about 3x. This could be due to a multitude of reasons.

# 3/6/26 analysis begins here

In [ ]:
print('HEALTH' in df_filtered2015.columns) # checking if same columns are present in 2015 dataset as in 2024 dataset
print('SPDPSTYR' in df_filtered2015.columns)
print('SEDRSSLEP' in df_filtered2015.columns)
print('TRQRSSLEP' in df_filtered2015.columns)
print('UDSTWDSLEEP' in df_filtered2015.columns)
print('UDSVWDSLEEP' in df_filtered2015.columns)

In [17]:
key_cols = ['AGE3', 'STMNMREC', 'SEDNMREC', 'HEALTH', 'SPDPSTYR', 'SEDRSSLEP', 'TRQRSSLEP', 'UDSTWDSLEEP', 'UDSVWDSLEEP'] # updating key columns to include sleep and health variables for 2024 dataset
df_filtered2024 = df_filtered2024[key_cols]
df_filtered2024 = df_filtered2024[df_filtered2024['AGE3'].between(3, 6)] #2024 dataset already filtered to ages 16-25, but reapplying filter after adding new columns just in case

print(df_filtered2024.shape)
print(df_filtered2024.head())

df_older = df_full_2024[df_full_2024['AGE3'].between(7, 8)]

print(df_older['STMNMREC'].value_counts())
print(df_older['SEDNMREC'].value_counts())

(18011, 9)
   AGE3  STMNMREC  SEDNMREC  HEALTH  SPDPSTYR  SEDRSSLEP  TRQRSSLEP  \
2     3        91        91       5       NaN         91         91   
3     4        91        91       3       0.0         91         91   
4     3        91        91       4       NaN         91         91   
5     6        91        91       2       0.0         91         91   
7     5        91        91       1       0.0         91         91   

   UDSTWDSLEEP  UDSVWDSLEEP  
2           91           91  
3           91           91  
4           91           91  
5           91           91  
7           91           91  
STMNMREC
91    8789
3      521
2      181
1       72
98      47
83      10
9        1
Name: count, dtype: int64
SEDNMREC
91    9337
3      190
98      46
2       20
83      15
1       12
9        1
Name: count, dtype: int64


In [ ]:
print(df_filtered2024['HEALTH'].value_counts())
print(df_filtered2024['SPDPSTYR'].value_counts())
print(df_filtered2024['SEDRSSLEP'].value_counts())
print(df_filtered2024['TRQRSSLEP'].value_counts())

In [13]:
print(df_filtered2024[df_filtered2024['STMNMREC'].isin([1,2])]['SPDPSTYR'].value_counts()) #checking for correlation between psychological stress and stimulant misuse in 2024 dataset
print(df_filtered2024[df_filtered2024['SEDNMREC'].isin([1,2])]['SPDPSTYR'].value_counts()) # same thing but for sedatives

SPDPSTYR
1.0    199
0.0    152
Name: count, dtype: int64
SPDPSTYR
1.0    31
0.0     8
Name: count, dtype: int64


### An interesting finding
199 / 408 who misuse stimulants have psychological stress.
152 / 408 did not have psychological stress

31 / 45 who misuse sedatives have psychological distress
8 / 45 did not have psychological distress


In [14]:
print(df_filtered2024[df_filtered2024['STMNMREC'].isin([1,2])]['HEALTH'].value_counts())
print(df_filtered2024[df_filtered2024['SEDNMREC'].isin([1,2])]['HEALTH'].value_counts())

HEALTH
2    163
3    131
1     57
4     50
5      7
Name: count, dtype: int64
HEALTH
2    14
4    13
3    13
5     4
1     1
Name: count, dtype: int64


In [ ]:
key_cols = ['AGE3', 'STMNMREC', 'SEDNMREC', 'HEALTH', 'SPDPSTYR'] #checking higher age ranges to decide if survey stigma might be affecting 
# reporting of stimulant and sedative misuse in younger age groups,
# by looking at psychological stress and health variables which might be correlated with substance misuse
df_2024older = df_full_2024[key_cols]
df_2024older = df_full_2024[df_2024older['AGE3'].between(7, 8)]

print(df_2024older['STMNMREC'].value_counts())
print(df_2024older['SEDNMREC'].value_counts())

STMNMREC
91    8789
3      521
2      181
1       72
98      47
83      10
9        1
Name: count, dtype: int64
SEDNMREC
91    9337
3      190
98      46
2       20
83      15
1       12
9        1
Name: count, dtype: int64


In [16]:
print(df_2024older[df_2024older['STMNMREC'].isin([1,2])]['SPDPSTYR'].value_counts())
print(df_2024older[df_2024older['SEDNMREC'].isin([1,2])]['SPDPSTYR'].value_counts())
#psycholigcal distress with higher age groups. and correlation to stimulant and sedative misuse

SPDPSTYR
0.0    143
1.0    110
Name: count, dtype: int64
SPDPSTYR
0.0    21
1.0    11
Name: count, dtype: int64


Stimulant abuse
Ages 16-25: 199 yes, 152 no — distress is more common 
Ages 26-35: 110 yes, 143 no — distress is less common

Sedative abuse
Ages 16-25: 31 yes, 8 no — strongly distressed
Ages 26-35: 11 yes, 21 no — more are not distressed